rustls/src/msgs/deframer.rs

```diff
@@ -397,6 +403,34 @@ impl DeframerVecBuffer {
     }
 }
 
+/// A borrowed version of [`DeframerVecBuffer`] that tracks discard operations
+pub struct DeframerSliceBuffer<'a> {
```

Comment:
I find it confusing that this gets introduced as part of a commit that describes itself as "batch discard operations", which seems not very related? Can we split these two things in separate commits?

Reply:
fixed